syntax = "proto3";


service BookStore {
  rpc CreateLocalStores(CreateLocalStoresRequest) returns (CreateLocalStoresResponse) {}
}

message CreateLocalStoresRequest {
  int32 processes_number = 1;
}

message CreateLocalStoresResponse {
  bool success = 1;
  string message = 2;
  string client_id = 3;
  repeated string processes_ids = 4;
  repeated string processes_ports = 5;
}



service DataStorage {
  rpc InitialWriteOperation(InitialWriteOperationRequest) returns (InitialWriteOperationResponse) {}
  rpc CommonWriteOperation(CommonWriteOperationRequest) returns (CommonWriteOperationResponse) {}
  rpc ListBooks(ListBooksRequest) returns (ListBooksResponse) {}
}

message InitialWriteOperationRequest {
  string book_name = 1;
  float book_price = 2;
}

message InitialWriteOperationResponse {
  bool success = 1;
  string message = 2;
}

message CommonWriteOperationRequest {
  string book_name = 1;
  float book_price = 2;
}

message CommonWriteOperationResponse {
  bool success = 1;
  string message = 2;
}

message ListBooksRequest {
}

message ListBooksResponse {
  bool success = 1;
  string message = 2;
}

